akka-stream/src/main/scala/akka/stream/impl/io/InputStreamPublisher.scala

```diff
@@ -47,7 +47,7 @@ private[akka] class InputStreamPublisher(is: InputStream, bytesReadPromise: Prom
   def readAndSignal(): Unit =
     if (isActive) {
       readAndEmit()
```

Comment:
It is too much effort to switch contexts right now, but if you think there is an issue/improvement possibility here then please open a ticket. I am busy backporting now :)